upstairs/src/lib.rs

```diff
@@ -4790,19 +4807,52 @@ impl Downstairs {
     /// # Panics
     /// If the different downstairs have different extents under repair (which
     /// is not allowed)
-    fn get_extent_under_repair(&self) -> Option<u64> {
-        let mut extent_under_repair = None;
+    fn get_extent_under_repair(&self) -> Option<std::ops::RangeInclusive<u64>> {
+        let mut first = None;
         for cid in 0..3 {
             if let Some(eur) = self.extent_limit[cid] {
-                if extent_under_repair.is_none() {
-                    extent_under_repair = Some(eur as u64);
+                if first.is_none() {
+                    first = Some(eur as u64);
                 } else {
                     // We only support one extent being repaired at a time
-                    assert_eq!(Some(eur as u64), extent_under_repair);
+                    assert_eq!(Some(eur as u64), first);
                 }
             }
         }
-        extent_under_repair
+        first.map(|first| {
+            first
+                ..=self
+                    .repair_job_ids
+                    .last_key_value()
+                    .map(|(k, _)| *k)
+                    .unwrap_or(first)
+        })
+    }
+
+    fn reserve_repair_jobs_for(&mut self, impacted_blocks: ImpactedBlocks) {
```

Comment:
Good question!  I decided to rephrase `reserve_repair_ids` and `reserve_repair_jobs` (very ambiguous naming!) into `check_repair_ids_for_extent` and `reserve_repair_ids_for_range`.

This is done in 39a8ac3047932990dff8eff18b43b646d532fa2e.

I made one other change while editing these functions: `reserve_repair_ids_for_extent` now _does not return_ the repair IDs.  This is because returning repair IDs without dependencies is questionable: if you're using those IDs, you _really should_ be using the dependencies as well (or at least binding + discarding the value).

The only places using the returned values were unit tests, so I adjusted those tests to check the IDs directly in `repair_job_ids` instead.

Reply:
Okay, I follow how that moved around, yeah.